### ISSUE TITLE
Exercise 8.3 solution 

### ISSUE THREAD

**Issue by amoat7** [ageron/handson-ml#402]
Please why is it necessary to sort the class indices in your fit function?

**Comment by ageron** [comment]
Hi @amoat7 ,
Are you sure it's exercise 8.3? I can't find what you are referring to.

**Comment by amoat7** [comment]
I'm sorry. please It's exercise no. 8, chapter 11.

**Comment by ageron** [comment]
No worries, thanks for clarifying! Well it's just to have a deterministic order for the class labels. It is quite a common practice, for example Scikit-Learn does the same thing. For example, suppose I train a `LogisticRegression` model to classify some data (say images of animals) like this:

```python
from sklearn.linear_model import LogisticRegression
model = LogisticRegression()
model.fit([[1, 2, 3], [4, 5, 6], [7, 8, 9], [10, 11, 12]], ["fox", "cat", "cat", "dog"])
```

When I make predictions with this model, I will get one probability per class:

```pycon
>>> model.predict_proba([[9, 10, 11]])
array([[0.50477365, 0.48073074, 0.01449561]])
```

What classes do these probabilities correspond to? Are these the probabilities of the fox, cat and dog classes respectively? This might make sense as it is the order in which I first used each class in the training set. But if the training set was shuffled before training (which is common practice), then it would make it impossible to know in advance what the order of the class probabilities in the predictions will be. So instead, Scikit-Learn orders the classes by alphabetical order: the first probability is for the cat, then the dog, then the fox. This way it does not matter how the training set is shuffled. You can see this by looking at the model's `classes_` attribute:

```pycon
>>> model.classes_
model.classes_
array(['cat', 'dog', 'fox'], dtype='<U3')
```

Hope this helps!

**Comment by amoat7** [comment]
Thank you
